src/components/sketches/Scene4.ts

```diff
@@ -0,0 +1,170 @@
+import p5 from 'p5';
+import {
+  BUBBLE_FREQUENCY,
+  BUBBLE_FREQUENCY_ON_LINES,
+  TRANSPARENCY,
+  TRANSPARENCY_RADIUS,
+  INC,
+  INC_START,
+  MAG_INC,
+  SCL,
+  NUM_PARTICLES,
+  STROKE_WEIGHT,
+} from '../../utils/constants';
+
+export default function sketch(p: p5): void {
+  let canvas: p5.Renderer;
+
+  let start = 0;
+  let cols: number;
+  let rows: number;
+  let zoff = 0;
+  const particles: Particle[] = [];
+  let flowfield: p5.Vector[];
+  let magOff = 0;
+  const showField = false;
+
+  p.setup = () => {
+    canvas = p.createCanvas(window.innerWidth, window.innerHeight);
+    canvas.id('p5-background');
+    p.pixelDensity(1);
+    cols = p.floor(window.innerWidth / SCL);
+    rows = p.floor(window.innerHeight / SCL);
+    p.background(0);
+
+    for (let i = 0; i < NUM_PARTICLES; i++) {
+      particles[i] = new Particle();
+    }
```

Comment:
nit: convert to map

Reply:
```		particles = Array(NUM_PARTICLES).fill(0).map(() => {
      return new Particle();
    });```
shouldnt this work? it's not..

Reply:
Particles needs to be declared with let not const

Reply:
ive tried many different syntaxes to get map to work, and none of them are working even spread operator for a deep copy, fill, new, idk how to fix it

Reply:
@ivanma9 

```
const particles: Particle[] = [];
```

this const on line 22 should be the problem as is said in the previous comment

change to:

```
let particles: Particle[] = [];
```

Reply:
yeet i changed it back for linters just before recent push, but when I was testing out different map methods i used `let` instead of const.

Reply:
basically, it still doesnt work with `let`